lab-3.py

```diff
@@ -0,0 +1,29 @@
+import subprocess
+import sys
+import boto3
+import datetime
+from shlex import quote
+
+s3 = boto3.client('s3')
+
+def main(argv):
+    cmd = argv
+    log_file_name = datetime.datetime(tzinfo=datetime.timezone.utc).now().strftime("%m_%d_%Y") + "_logfile"
+    kickoff_subprocess(cmd, log_file_name)
+    upload_output_to_S3(log_file_name)
+
+def kickoff_subprocess(cmd, log_file_name):
+    process = subprocess.call(quote(cmd), shell=True)
+    file = open(log_file_name, "a+")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *file*. The resource is allocated by call *builtins.open*. The resource is referenced at lines 20, 21. The resource is closed at line 21. There are other execution paths that do not contain closure statements, for example, when *TextIOWrapper.write()* throws an exception. To prevent this resource leak, close *file* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).